sh_script/fuzzing.sh

```diff
@@ -156,10 +160,12 @@ run_single_case() {
             fi
 		    timeout $test_time cargo afl fuzz -i fuzz/seeds/${test_case#*_}/ -o fuzz/artifacts/$test_case fuzz/target/debug/$test_case
             
-            if [ -d "${test_case}_cov" ]; then
-                rm -rf "${test_case}_cov"
+            if [ "${collect_coverage}" == "YES" ]; then
+                if [ -d "${test_case}_cov" ]; then
+                    rm -rf "${test_case}_cov"
```

Comment:
Nit: This could be simplified to:

```suggestion
                [ -d "${test_case}_cov" ] && rm -rf "${test_case}_cov"
```